### ISSUE TITLE
UI improvement for segment folder menu?

### ISSUE THREAD

**Issue by ethanbass** [Abe404/root_painter#114]
The segment folder menu currently makes you select the input and output directories manually even though (as far as I can tell), it is sharply constrained by the program what folders you can use for this. I believe the output directory has to be the "segmentation" directory and the input directory has to be the image directory that was already specified when opening the project. It would be convenient (and less confusing) if these were filled in automatically from the project file. Also, if it's easy to do with the widget, I think the model files could be in a drop down menu that pulls up the files from the `models` folder as options instead of having to navigate the whole file system.
All best,
Ethan

**Comment by Abe404** [comment]
Hi Ethan,


I agree there’s some room for improvement in terms of more sensible defaults for these options.

> I believe the output directory has to be the "segmentation" directory

This is not currently the case. My personal suggestion is to create a “results” folder within each project folder that contains a segmentation folder within it that perhaps also includes the model number in the folder name. For example projects/project_name/results/seg_model_10

Then when you segment a dataset with a specific model you can add the segmentations to the project (in its results folder) which was used for training that model. This is just my suggestion.

The projects/project_name/segmentations folder exists for every project and includes the segmentations created as part of the interactive training procedure. I suggest leaving these as they are (don’t overwrite them) so they can be used for computing metrics etc. I suggest outputting additional segmentations (created when using the segment folder function) into your own user-defined results folder or similar.

I think it would help to have my suggestions as defaults when using the segment folder function from within an open project and also to automatically select the model with the highest number.

It’s currently possible to use the segment folder option without even opening a project. In that case it’s not clear which models, output folders or datasets would be selected. In that case I think there shouldn’t be any defaults.

I think things are a bit more open ended than you believed. Perhaps this helps explain the lack of defaults?

Kind regards,
Abraham

**Comment by Abe404** [comment]
Concretely: For segmenting folders with an open project, I believe model and output directory should be automatically suggested/populated.

Dataset is less clear due to the difference between the training dataset and original images. I think the user has to specify this for now but perhaps something to come back to later.

**Comment by ethanbass** [comment]
Yes this makes a lot of sense -- thank you for the clarification.  In light of your recommendations I'm even more convinced that it would be helpful to provide some sensible defaults for these parameters.  I'm sure the thing about the recommendation about results folder is well documented somewhere, but I think I overlooked it when I was reading through the docs. Perhaps it would make sense to automatically create the results folder when the project is created?

Re: the training dataset -- the way I've been dealing with this is by creating a separate project for the training data and then importing the trained model into a new project for doing the actual analyses, so I don't think I'm overwriting anything by writing the segmentations into the segmentations folder. Do you see any issue with doing it this way?

Thanks for all your insights!
Ethan
